k-distribution/src/main/scripts/lib/checkJava

```diff
@@ -46,15 +46,20 @@ else
   fi
 fi
 
-
+# The Java version information we're parsing here looks something like this:
+#
+#   openjdk 21.0.2 2024-01-16
+#   OpenJDK Runtime Environment Homebrew (build 21.0.2)
+#   OpenJDK 64-Bit Server VM Homebrew (build 21.0.2, mixed mode, sharing)
+#
+# To get the major version, we need to look at the first line's second field,
+# then use a regex to extract the major version component. If the JVM changes
+# the format of the version that gets printed, this script needs to be updated
+# as well.
 if [[ "$_java" ]]; then
```

Comment:
This condition on `_java` has been here since the file was created. But AFAICT, it will always evaluate to true. Any idea why it's here?

Reply:
Good point; there's one branch where it's not set at the top of the file, so maybe the intention was to leave it unset there and just emit the error later on?